purchase_order_report_selection/models/purchase_order.py

```diff
@@ -0,0 +1,20 @@
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    report_field_ids = fields.One2many("purchase.report.field", "purchase_order_id")
+
+
+class PurchaseReportField(models.Model):
+    _name = "purchase.report.field"
+    _description = "Purchase report field"
+
+    field = fields.Many2one(
+        "ir.model.fields", domain=[("model", "=", "purchase.order.line")]
```

Comment:
please, use separate method for domain, it will more inheritable:

```python
@api.model
def _domain_field_id(self):
    return [("model", "=", "purchase.order.line")]

field_id = fields.Many2one("ir.model.fields", domain=lambda self: self._domain_field_id())
```

Reply:
Understood, I will keep in mind